app/frontend/app/components/post-comment.js

```diff
@@ -1,23 +1,41 @@
 import Ember from 'ember';
 
 export default Ember.Component.extend({
-    classNames: ["status-update-panel"],
+  classNames: ["status-update-panel"],
+  formIsOpen: false,
 
   didInsertElement: function() {
     var self = this;
     this.$(".status-form").focus(function() {
-      self.$(".status-form").autosize({append: "\n"});
-      self.$(".panel-footer").slideDown(200);
+      self.set('formIsOpen', true);
+      self.handleFormState();
     });
     this.$(".status-form").blur(function() {
       if (self.$(".status-form").val().replace(/\s/g, '').length === 0) {
+        self.set('formIsOpen', false);
+        self.handleFormState();
+      }
+    });
+    this.$("#status-box-text").click(function() {
+      self.toggleProperty('newPostAdult');
+    });
+  },
+
+  handleFormState: function() {
+    var self = this;
+
+    setTimeout(function() {
+      if(self.get('formIsOpen')) {
+        self.$(".status-form").autosize({append: "\n"});
+        self.$(".panel-footer").slideDown(200);
+      } else {
         self.$(".status-form").val('');
         self.$(".panel-footer").slideUp(200, function() {
           self.$(".status-form").trigger("autosize.destroy");
         });
       }
-    });
-  },
+    }, 300);
```

Comment:
What's up with the `setTimeout` here? Is it a workaround for a bug?  If so, you should probably add a comment explaining that